### PR TITLE
Support Android Single Pass instead of unavailable Single Pass Instanced

### PR DESCRIPTION
**Pull request by Alexees**
## Overview
The Android Platform does not support Single Pass Instanced as its rendering mode, but the the optimization window tries to test it.
This PR adds a secondary preprocessored code path for Android specificially, only for test Single Pass on that platform.

## Changes
- Fixes: #7518 .

### CODE REVIEW

File: Assets/MRTK/Core/Inspectors/Setup/MixedRealityEditorSettings.cs [hunk 1]
```diff
@@ -98,10 +98,17 @@ private static void LogConfigurationWarnings()
                 Debug.LogWarning("<b>Virtual reality supported</b> not enabled. Check <i>XR Settings</i> under <i>Player Settings</i>");
             }
 
+#if UNITY_ANDROID
+            if (!MixedRealityOptimizeUtils.IsSinglePass())
```
Comment (davidkline-ms) [inline Assets/MRTK/Core/Inspectors/Setup/MixedRealityEditorSettings.cs]: Maybe use the same IsSinglePassInstanced, etc and move the UNITY_ANDROID checks inside those methods?

File: Assets/MRTK/Core/Inspectors/Setup/MixedRealityProjectConfiguratorWindow.cs [hunk 2]
```diff
@@ -17,7 +17,11 @@ public class MixedRealityProjectConfiguratorWindow : EditorWindow
             { MRConfig.ForceTextSerialization, true },
             { MRConfig.VisibleMetaFiles, true },
             { MRConfig.VirtualRealitySupported, true },
+#if UNITY_ANDROID
+            { MRConfig.SinglePass, true },
```
Comment (davidkline-ms) [inline Assets/MRTK/Core/Inspectors/Setup/MixedRealityProjectConfiguratorWindow.cs]: Recommend reusing the SinglePassInstancing enum value,

Comment (davidkline-ms) [reply Assets/MRTK/Core/Inspectors/Setup/MixedRealityProjectConfiguratorWindow.cs]: Maybe define SinglePass as the same value and mark SinglePassInstancing as obsolete.

Comment (Alexees) [reply Assets/MRTK/Core/Inspectors/Setup/MixedRealityProjectConfiguratorWindow.cs]: Not exactly sure what you mean by that. Is it possible to mark an enum value as obsolete?

Comment (Troy-Ferrell) [reply Assets/MRTK/Core/Inspectors/Setup/MixedRealityProjectConfiguratorWindow.cs]: I think @davidkline-ms  is saying make a new enum like
MRConfig.OptimalRenderingPath 

and then these get/set would do the appropriate android macro checking
and deprecrate the MRConfig.SingelPassInstancing. which yes you can put obsolete tag under just that item in the enum

File: Assets/MRTK/Core/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs [hunk 3]
```diff
@@ -100,7 +104,11 @@ public bool IsActiveBuildTargetValid()
             { Configurations.ForceTextSerialization, new ConfigGetter(() => { return IsForceTextSerialization(); }) },
             { Configurations.VisibleMetaFiles, new ConfigGetter(() => { return IsVisibleMetaFiles(); }) },
             { Configurations.VirtualRealitySupported, new ConfigGetter(() => { return XRSettingsUtilities.LegacyXREnabled; }) },
-            { Configurations.SinglePassInstancing, new ConfigGetter(() => { return MixedRealityOptimizeUtils.IsSinglePassInstanced(); }) },
+#if UNITY_ANDROID
+            { Configurations.SinglePass,  new ConfigGetter(() => { return MixedRealityOptimizeUtils.IsSinglePass(); }) },
+#else
+            { Configurations.SinglePassInstancing,  new ConfigGetter(() => { return MixedRealityOptimizeUtils.IsSinglePassInstanced(); }) },
+#endif
```
Comment (Troy-Ferrell) [inline Assets/MRTK/Core/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs]: per comment above, change to 

{ Configurations.OptimalRenderingPath, new CONfigGetter(() =>
{
if Android
do blah
else
do regular
endif
}};

File: Assets/MRTK/Tools/OptimizeWindow/MixedRealityOptimizeWindow.cs [hunk 4]
```diff
@@ -395,7 +399,11 @@ private void RenderSettingOptimizations()
                 EditorGUILayout.LabelField(ToolbarTitles[(int)ToolbarSection.Settings], MixedRealityStylesUtility.BoldLargeTitleStyle);
                 using (new EditorGUI.IndentLevelScope())
                 {
+#if UNITY_ANDROID
                     RenderSinglePassSection();
+#else
+                    RenderSinglePassInstancingSection();
+#endif
```
Comment (Troy-Ferrell) [inline Assets/MRTK/Tools/OptimizeWindow/MixedRealityOptimizeWindow.cs]: I would also change this to 
RenderOptimalRenderingSection or DrawOptimalRenderingPathSection etc

File: Assets/MRTK/Core/Inspectors/Setup/MixedRealityProjectConfiguratorWindow.cs (hunk 2 above)
Comment (davidkline-ms) [reply Assets/MRTK/Core/Inspectors/Setup/MixedRealityProjectConfiguratorWindow.cs]: The suggestion is to set the value of SinglePass to that of SinglePassInstanced. For example (please do not use these values, they are examples)

```
{
   ...
   SinglePass = 17,
   [Obsolete("SinglePassInstancing is obsolete, use SinglePass instead")]
   SinglePassInstancing = 17,
   ...
}
```

File: Assets/MRTK/Core/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs [hunk 5]
```diff
@@ -30,7 +30,9 @@ public enum Configurations
             ForceTextSerialization,
             VisibleMetaFiles,
             VirtualRealitySupported,
-            SinglePassInstancing,
+            [Obsolete("SinglePassInstancing is obsolete, use SinglePass instead")]
```
Comment (keveleigh) [inline Assets/MRTK/Core/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs]:
```suggestion
            [Obsolete("SinglePassInstancing is obsolete, use OptimalRenderingPath instead")]
```

**Comment by davidkline-ms** [comment]
@Alexees, merging #7533 caused a merge conflict. If you can fix that, we can merge this one. Thanks!

**Pull request merged**